### ISSUE TITLE
Error on withdrawing multiple statecoins - coins in limbo

### ISSUE THREAD

**Issue by cbjd** [layer2tech/mercury-wallet#916]
Attempted to withdraw two statecoins simultaneously, an RPC error was thrown:  

```
Transaction broadcast failed with error: Error: 400 - "sendrawtransaction RPC error: {\"code\":-25,\"message\":\"bad-txns-inputs-missingorspent\"}".
```

I was alerted to check the withdrawn statecoins pane to grab the raw hex. When looking at the raw transaction, it seems to be spending from a nonexistent UTXO, and the script_type is empty. I can share the full raw transaction out of band if that is helpful. I am no longer able to access the coins. 

Statecoin addresses:
sc1q25vcr8h0gn7xc575ygz8d6qjxrwjryfqczxy6mgqa7cwe5gceuuvlg9ypy
sc1qdfu48v4pj3a0jykfcyzmlx5eyjmx7pcsrwh6h4c646rutjheeklyw2zg99

**Comment by tomt1664** [comment]
Yes, please share the tx hex - will need to know the outputs to figure out what's happened here.

**Comment by cbjd** [comment]
Will share it here, these are coins I swapped for so I don't care if they are public (lol). 

```
020000000001025053c0a958af7637d6cd13928b1212da48fb4e5ad6e977654aef1b9ef8c73a2e0000000000ffffffff6f993acf9b4daad419ccd74e7f9f6fd1bd8936c1983d03245a93c5933c3a8b4e0000000000ffffffff02df050300000000001600141986a3a56ca8cece151ae50add2b0b5256d51851e8030000000000001600147a0d5c4648afd38b35f45bffecc5d5dc423bda010247304402205efe037f76f818342a88f3ecab614877ee0b6c4192df85afdf70dcaafd8e6d01022035eab590b1ef2a85c68253b7c733b589be4174505dc1fef8b87acf652a41cc91012103d4f6b61e82f24bbfd7ac5c4f5847a3123eefdb362bc81b05a71d9550b67a60e50247304402205a6a4ef2d14c8651de1aa359472ebb247aa3e6f0a062b0bc833a9d8ca7ffe469022026e48b38754e012287f1855cc1879137d92dd706b030429d3f236d552d6d00af012103aac3286b5f4bfac300f8ddf9814370812ee7bfec70f1034144596398f5a2373e00000000
```

UTXOs:
```
01220dc38b6fb5568e9ade7384b9c9720d48196cb498d4087f2e276f093a3bb3:0
2e3ac7f89e1bef4a6577e9d65a4efb48da12128b9213cdd63776af58a9c05350:0
```

**Comment by tomt1664** [comment]
The cause of this (RBF during deposit) has been fixed in v0.4.54. 

Leaving this issue open though until we get your coins back.

**Comment by takinbo** [comment]
Is there an update on this issue? Had the exact same issue and after checking the utxos, I can confirm that they all have RBF signaled.

**Comment by tomt1664** [comment]
Do you have coins affected by this (i.e. deposited last year and and the deposit transaction was replaced before deposit)? We have a recent update that enables the recovery of accidentally deposited coins - I need to do some checks on the procedure to recover in this case - I will get back to you. 

If you are happy to share the coin info, I can check the status.

**Comment by tomt1664** [comment]
@takinbo The latest release (0.6.28 https://github.com/layer2tech/mercury-wallet/releases/tag/v0.6.28) has a modification that will enable withdrawal of any coin with an incorrect deposit outpoint (from e.g. a replacement deposit tx with an old version of the wallet). 

With this version, load in the wallet that contained the stuck coin. Then go to the settings page and click on 'Check for duplicate coins' button. Any coin with an incorrect deposit txid (even if it had been 'withdrawn') will now appear as a 'duplicate' status coin and can be withdrawn. 

Let us know if you have any issues with this.

**Comment by takinbo** [comment]
That's great news! It seems the release (0.6.28) includes detached signatures from a previous version (0.6.21) of the wallet so I'm unable to verify its integrity. Could you take a look?

**Comment by tomt1664** [comment]
Thanks for spotting that - updated with the correct sigs now.

**Comment by takinbo** [comment]
Following up on this issue. I followed the steps outlined and was unable to retrieve any deposits. It makes sense since I did not deposit the utxos that I was attempting to withdraw; I got them from swaps that I had made earlier. Any suggestions on what else I could try? I'm happy to share more details about the withdrawal transaction privately.

**Comment by tomt1664** [comment]
If there was an incorrect txid due to an RBF during deposit, then that coin should not have been able to be received either from a transfer or a swap (as the wallet verifies the coin). Can you please share some more details - you can send to main@mercurywallet.com. If you can send the coin details (txid:vout) as well as backup tx (Settings -> Manage backup txs -> copy raw hex) and the withdrawal tx raw hex (filter for withdrawn coins and then select coin and copy raw hex) that would help. Thanks.

**Comment by Mentakh** [comment]
Just wanted to add that this thread helped me retrieve my duplicate coins after accidentally depositing twice to the same statecoin.

I "checked for duplicate coins" in settings, and the duplicate coins were displayed in red. Had to withdraw the original statecoins and then the duplicates one by one. Worked perfectly.
